clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -362,14 +368,32 @@ bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& re
     processNextCall();
   }
 
+  bool interrupted = false;
   {
     boost::mutex::scoped_lock lock(info->finished_mutex_);
 
     while (!info->finished_)
     {
-      info->finished_condition_.wait(lock);
+      if (timeout > ros::WallDuration(0.0))
+      {
+        if (info->finished_condition_.wait_for(lock, boost::chrono::nanoseconds(timeout.toNSec())) == boost::cv_status::timeout)
+        {
+          ROS_WARN("Service [%s] call failed: no response for %fsec", service_name_.c_str(), timeout.toSec());
+          interrupted = true;
```

Comment:
The `_success` that is returned doesn't seem to be set to `false`, or am I missing something here?

Reply:
If I'm not wrong, it's initialized to `false` here: https://github.com/efernandez/ros_comm/blob/14b168bfbec4c1a2fd74302836dc2247fa9317a1/clients/roscpp/src/libros/service_server_link.cpp#L340

And set to `true/false` if the call is attended here: https://github.com/efernandez/ros_comm/blob/14b168bfbec4c1a2fd74302836dc2247fa9317a1/clients/roscpp/src/libros/service_server_link.cpp#L213